sahi/slicing.py

```diff
@@ -466,3 +469,205 @@ def slice_coco(
+        orientation, slide=1, ratio=0.8
+    )
+
+    slice_height = height // split_col  # noqa
+    slice_width = width // split_row  # noqa
+
+    x_overlap = int(slice_width * overlap_width_ratio)  # noqa
+    y_overlap = int(slice_height * overlap_height_ratio)  # noqa
+
+    return x_overlap, y_overlap, slice_width, slice_height
+
+
+def upper_medium_resolution(height, width, orientation):
+    """
+
+    Args:
+        height:
+        width:
+        orientation: image capture orientation
+
+    Returns:
+        overlap params
+    """
+    split_row, split_col, overlap_height_ratio, overlap_width_ratio = calc_ratio_and_slice(
+        orientation, slide=2, ratio=0.4
+    )
+
+    slice_height = height // split_col  # noqa
+    slice_width = width // split_row  # noqa
+
+    x_overlap = int(slice_width * overlap_width_ratio)  # noqa
+    y_overlap = int(slice_height * overlap_height_ratio)  # noqa
+
+    return x_overlap, y_overlap, slice_width, slice_height
+
+
+def high_resolution(height: int, width: int, orientation: str):
+    """
+
+    Args:
+        height:
+        width:
+        orientation: image capture orientation
+
+    Returns:
+        overlap params
+    """
+    split_row, split_col, overlap_height_ratio, overlap_width_ratio = calc_ratio_and_slice(
+        orientation, slide=4, ratio=0.4
+    )
+
+    slice_height = height // split_col  # noqa
+    slice_width = width // split_row  # noqa
+
+    x_overlap = int(slice_width * overlap_width_ratio)  # noqa
+    y_overlap = int(slice_height * overlap_height_ratio)  # noqa
+
+    return x_overlap, y_overlap, slice_width, slice_height
+
+
+def resolution_selector(res: str, height: int, width: int):
+    """
+
+    Args:
+        res: resolution of image such as low, medium
+        height:
+        width:
+
+    Returns:
+        trigger slicing params function and return overlap params
+    """
+    orientation = cal_aspect_ratio_orientation(width=width, height=height)
+    resolution_select = {
+        "low": low_resolution(height=height, width=width),
+        "medium": medium_resolution(height=height, width=width, orientation=orientation),
+        "upper_medium": upper_medium_resolution(height=height, width=width, orientation=orientation),
+        "high": high_resolution(height=height, width=width, orientation=orientation),
+    }
+    return resolution_select.get(res)
+
+
+def calc_slice_params(height: int, width: int):
+    """
+    According to Image HxW calculate overlap sliding window and buffer params
+    factor is the power value of 2 closest to the image resolution.
+        factor <= 18: low resolution image such as 300x300, 640x640
+        18 < factor <= 21: medium resolution image such as 1024x1024, 1336x960
+        21 < factor <= 24: upper medium resolution image such as 2048x2048, 2048x4096, 4096x4096
+        factor > 24: high resolution image such as 6380x6380, 4096x8192
```

Comment:
I'm not sure about this as it may not reflect the true standards (but it approximates), but what about changing `upper medium -> high` and `high -> ultra high` ? @mcvarer @fcakyon